src/lib/AckDevProvider.tsx

```diff
@@ -0,0 +1,131 @@
+"use client";
+
+import {
+  createContext,
+  ReactNode,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import Dialog from "@mui/material/Dialog";
+import DialogTitle from "@mui/material/DialogTitle";
+import DialogContent from "@mui/material/DialogContent";
+import DialogContentText from "@mui/material/DialogContentText";
+import DialogActions from "@mui/material/DialogActions";
+import Button from "@mui/material/Button";
+import * as React from "react";
+import { usePathname, useSearchParams } from "next/navigation";
+import { useTranslations } from "next-intl";
+
+const IsADeveloperContext = createContext<[boolean, () => void]>([
+  false,
+  () => {},
+]);
+
+interface IsDeveloperProviderProps {
+  children: ReactNode;
+}
+
+const MESHFORMS_IS_A_DEVELOPER_ACK_TIME =
+  "meshforms_user_last_acked_developer_status";
+const DEVELOPER_ACK_CACHE_DURATION_MILLISECONDS = 1000 * 60 * 60 * 24 * 30; // 30 days
+
+export const IsDeveloperProvider: React.FC<IsDeveloperProviderProps> = ({
+  children,
+}) => {
+  const searchParams = useSearchParams();
+  const [showAskIfDeveloperDialog, setShowAskIfDeveloperDialog] =
+    useState<boolean>(false);
+  const [isADeveloper, setIsADeveloper] = useState<boolean>(false);
+  const t = useTranslations("DeveloperMode");
+
+  useEffect(() => {
+    if (isADeveloper) {
+      localStorage.setItem(
+        MESHFORMS_IS_A_DEVELOPER_ACK_TIME,
+        Date.now().toString(),
+      );
+    }
+
+    if (searchParams.get("isDeveloper") === "true") {
+      // Let bots like Datadog by pass the dialog with a query param
+      setIsADeveloper(true);
+      return;
+    }
```

Comment:
Clever. Will you make the appropriate adjustments to the synthetics?

Reply:
Done